tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/upgrade/FromSolrUpgradeTest.java

```diff
@@ -0,0 +1,192 @@
+package org.alfresco.elasticsearch.upgrade;
+
+import static java.time.Duration.ofMinutes;
+
+import static org.alfresco.elasticsearch.AlfrescoStackInitializer.getImagesConfig;
+
+import java.io.IOException;
+import java.net.URL;
+
+import org.alfresco.elasticsearch.EnvHelper;
+import org.alfresco.elasticsearch.upgrade.AvailabilityProbe.Stats;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class FromSolrUpgradeTest
+{
+    private static final URL TEST_FILE_URL = FromSolrUpgradeTest.class.getResource("babekyrtso.pdf");
+    private static final String SEARCH_TERM = "babekyrtso";
+    private static final String FILE_UPLOADED_BEFORE_INITIAL_REINDEXING = "before-initial-re-indexing.pdf";
+    private static final String FILE_UPLOADED_BEFORE_STARTING_LIVE_INDEXING = "before-live-indexing.pdf";
+    private static final String FILE_UPLOADED_AFTER_STARTING_LIVE_INDEXING = "after-live-indexing.pdf";
+    private static final String FILE_UPLOADED_BEFORE_SWITCHING_TO_ELASTICSEARCH = "before-switch.pdf";
+    private static final String FILE_UPLOADED_AFTER_SWITCHING_TO_ELASTICSEARCH = "after-switch.pdf";
+
+    @Test
+    public void testZeroDowntimeUpgradeFromSolrToElasticsearch() throws IOException
+    {
+        try (final UpgradeScenario scenario = new UpgradeScenario(getUpgradeScenarioConfig()))
+        {
+            final ACSEnv initialEnv = scenario.startInitialEnvWithSolrBasedSearchService();
```

Comment:
I'd try to split this test up with methods a bit, like a separate method "initialEnv = startInitialEnvironment()", "startMirroredEnvironment()", some methods like "verifyInitialEnvironmentIndexing()". It's a bit hard to read now when it's all on one level and we kind of lose the possibility of "documentation" through method names making it easier to read and separate code.

Reply:
To be honest I don't know how to split it without overengineering it or mixing high level code with low level one.

Reply:
They can git blame as we've talked to get the description of this case